src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -753,3 +863,66 @@ nlohmann::json ExportQueryExecutionTrees::computeResultAsJSON(
     throw;
   }
 }
+
+// _____________________________________________________________________________
+cppcoro::generator<std::string>
+ExportQueryExecutionTrees::computeResultAsQLeverJSONStream(
+    const ParsedQuery& query, const QueryExecutionTree& qet,
+    const ad_utility::Timer& requestTimer,
+    CancellationHandle cancellationHandle) {
+  std::shared_ptr<const Result> result = qet.getResult();
+  result->logResultSize();
+  auto timeResultComputation = requestTimer.msecs();
+
+  nlohmann::json jsonPrefix;
+
+  jsonPrefix["query"] = query._originalString;
+  jsonPrefix["status"] = "OK";
+  jsonPrefix["warnings"] = qet.collectWarnings();
+  if (query.hasSelectClause()) {
+    jsonPrefix["selected"] =
+        query.selectClause().getSelectedVariablesAsStrings();
+  } else {
+    jsonPrefix["selected"] =
+        std::vector<std::string>{"?subject", "?predicate", "?object"};
+  }
```

Comment:
As I see it, 
Most of this code is duplicated from the materializing exporter.
There's two ways to go here:

1. Completely remove the materializing exporters.
2. Factor out the common code for determining the `prefix` and `suffix`.

The same goes probably for the header of the `SparqlJSON` format.